src/api/v1/community_businesses/visit_activities.ts

```diff
@@ -0,0 +1,130 @@
+import * as Hapi from 'hapi';
+import * as moment from 'moment' ;
+import { CommunityBusinesses } from '../../../models';
+import { getCommunityBusiness } from '../prerequisites';
+import {
+  response,
+  visitActivitiesPostPayload,
+  visitActivitiesPutPayload,
+  id,
+  meOrId,
+  visitActivitiesGetQuery } from './schema';
+import { VisitActivity } from '../../../models/types';
+import { Day } from '../../../types/internal';
+
+interface GetRequest extends Hapi.Request {
+  query: {
+    day: Day
+  };
+}
+
+interface PostOrPutRequest extends Hapi.Request {
+  payload: Partial<VisitActivity>;
+}
+
+export default [
+  {
+    method: 'GET',
+    path: '/community-businesses/{communityBusinessId}/visit_activities',
+    options: {
+      description: 'Retrieve all visit activities for a community business',
+      auth: {
+        strategy: 'standard',
+        access: {
+          scope: [
+            'visit_activities-own:read',
+            'visit_activities-child:read',
+            'visit_activties-parent:read',
+          ],
+        },
+      },
+      pre: [
+        { method: getCommunityBusiness, assign: 'communityBusiness' },
+      ],
+      validate: {
+        query: visitActivitiesGetQuery,
+        params: { communityBusinessId: meOrId },
+      },
+      response: { schema: response },
+    },
+    handler: async (request: GetRequest, h: Hapi.ResponseToolkit) => {
+      const { knex, pre: { communityBusiness }, query: { day: _day = null } } = request;
+      const day = _day === 'today'
```

Comment:
If we treat `'today'` as a different case from all other values, and end up casting it to an explicit weekday value, surely it shouldn't be a part of the `Day` union type?

Reply:
_done_